endpoint.go

```diff
@@ -289,10 +290,14 @@ func (ep *endpoint) Join(sbox Sandbox, options ...EndpointOption) error {
 		return err
 	}
 
-	if err = sb.populateNetworkResources(ep); err != nil {
-		return err
+	sb.Lock()
```

Comment:
The comment is about handling the osSbox nil check inside `populateNetworkResources()` as it is still being done in `clearNetworkResources()`.

If it is possible to do ina  clean way, I feel it would help readability.